packages/@lwc/engine-core/src/framework/hot-swaps.ts

```diff
@@ -22,6 +22,14 @@ const activeTemplates = new WeakMap<Template, Set<VM>>();
 const activeComponents = new WeakMap<LightningElementConstructor, Set<VM>>();
 const activeStyles = new WeakMap<StylesheetFactory, Set<VM>>();
 
+function assertNotProd() {
+    /* istanbul ignore if */
+    if (process.env.NODE_ENV === 'production') {
+        // this method should never leak to prod
+        throw new ReferenceError();
+    }
+}
```

Comment:
I _could_ move these to `@lwc/shared`, but that feels like overkill for something that is supposed to be tree-shaken out in prod anyway.

Reply:
Since all these files are under `engine-core/src/framework`, you could put them into `./utils` if you wanted to consolidate.

Reply:
Good point! Consolidated.